SPEC.md

```diff
@@ -201,25 +201,161 @@ Note that logging mechanisms other than stdout and stderr are not required by th
 ### Isolators
 
 Isolators enforce resource constraints rather than namespacing.
-Isolators may be applied to individual applications, to whole containers, or to both.
+Isolators may be scoped to individual applications, to whole containers, or to both.
 Some well known isolators can be verified by the specification.
 Additional isolators will be added to this specification over time.
 
-|Name|Type|Schema|Example|
-|---------------------------|------|------------------------------------|------------------------------------|
-|cpu/shares                 |string|"&lt;uint&gt;"                      |"4096"                              |
-|memory/limit               |string|"&lt;bytes&gt;"                     |"1G", "5T", "4K"                    |
-|cpu/mask                   |string|"&lt;cpu range&gt;"                 |"0", "0-3", "0,2,4-7"               |
-|block-io/read-bandwidth    |string|"&lt;path to file&gt; &lt;bytes&gt;"|"/tmp 1K"                           |
-|block-io/write-bandwidth   |string|"&lt;path to file&gt; &lt;bytes&gt;"|"/tmp 1K"                           |
-|network-io/read-bandwidth  |string|"&lt;device name&gt; &lt;bytes&gt;" |"eth0 100M"                         |
-|network-io/write-bandwidth |string|"&lt;device name&gt; &lt;bytes&gt;" |"eth0 100M"                         |
-|capabilities/bounding-set  |string|"&lt;cap&gt; &lt;cap&gt; ..."       |"CAP_NET_BIND_SERVICE CAP_SYS_ADMIN"|
+An isolator is a standalone JSON object with only one required field: "name".
+All other fields are specific to the isolator.
 
-#### Types
+An executor MAY ignore isolators that it does not understand and run the container without them.
+But, an executor MUST make information about which isolators were ignored, enforced or modified to the user.
+An executor MAY implement a "strict mode" where an image cannot run unless all isolators are in place.
 
-* uint: base 10 formatted unsigned int as a string
-* bytes: Suffix to a base 10 int to make it a K, M, G, or T for base 1024
```

Comment:
As much as I hate the kibibytes junk, some resources are NATURALLY base 10
and some are base 2.  Disk is base 10.  Network is base 10.  Memory is base
2.

For kubernetes we sucked it up and distinguished k (I am pretty sure SI
says k not K) from Ki and M from Mi

On Thu, Feb 19, 2015 at 10:53 PM, Charles D. Aylward <
notifications@github.com> wrote:

> In SPEC.md <https://github.com/appc/spec/pull/192#discussion_r25051779>:
>
> >
> > -* uint: base 10 formatted unsigned int as a string
> > -* bytes: Suffix to a base 10 int to make it a K, M, G, or T for base 1024
>
> Agreed, specifying which prefixes should be used should remain (e.g.
> binary-prefixes, K = 1024). And, please don't mention kibibytes. ;)
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/spec/pull/192/files#r25051779>.
>